### DISCUSSION TITLE
Resize the UML shapes dynamically based on the text length

### DISCUSSION THREAD

**Discussion by aylafiryal** [clientIO/joint#2212] [Q&A] [answered]
### Introduction

I encountered a problem while dealing with class diagram elements. The size of the elements doesn't increase when the text is long, as shown in the attached photo. Should I use the calc() function or the CustomTextElement from [the relative dimensions article](https://resources.jointjs.com/tutorial/special-attributes)? If so, how can I implement it in the UML element?

This is how I create the UML class. I'm sure the problem is that because I set the size of the shape to be 220 x 100 and that size doesn't match the size of my text box

`new uml.Class({`
`                position: { x:20  , y: 190 },`
`                size: { width: 220, height: 100 },`
`                name: Alya,`
`                attributes: atribut,`
`                methods: method,`
`               attrs: {`
`                    // some attrs`
`                }`
`})`

![image](https://github.com/clientIO/joint/assets/83856972/6d0b6d14-a5d2-4c20-80e1-f453d06eaa3d)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
*Note: It's an old shape and it will be removed in version `v4.0`.*

If you want the UML shape to wrap around all the text you can resize it based on the number of `methods` and `attributes`. For instance:

```js
graph.getElements().forEach(function(el) {
    const size = (el.get('methods').length + el.get('attributes').length) * 20 + 30;
    el.resize(220, size);
});
```

Otherwise, you could create an UML shape of your own following this tutorial: https://resources.jointjs.com/tutorial/content-driven-element

Or use the `Record` shape from JointJS+: 
- https://www.jointjs.com/demos/uml-class-shape-inspector
- https://www.jointjs.com/demos/uml-class-diagrams

**Reply by aylafiryal** [reply]
This really worked out well, thank you so much for taking the time to answer my question. I will consider using Joinjs+ later
